packages/iov-bns-governance/src/governor.ts

```diff
@@ -0,0 +1,109 @@
+import BN = require("bn.js");
+
+import { Address, Identity, WithCreator } from "@iov/bcp";
+import {
+  bnsCodec,
+  BnsConnection,
+  BnsTx,
+  CreateProposalTx,
+  Electorate,
+  Proposal,
+  TallyTx,
+  VoteOption,
+  VoteTx,
+} from "@iov/bns";
+import { Encoding } from "@iov/encoding";
+
+import { CommitteeId, ProposalOptions, ProposalType } from "./proposals";
+
+/* Constants to be finalised once the committees are actually on-chain */
+export const COMMITTEE_IDS = {
+  testnet: {
+    GOVERNING_BOARD: 1 as CommitteeId,
+    ECONOMIC_COMMITTEE: 2 as CommitteeId,
+    VALIDATOR_COMMITTEE: 3 as CommitteeId,
+    TECH_COMMITTEE: 4 as CommitteeId,
+  },
+};
+
+export interface GovernorOptions {
+  readonly connection: BnsConnection;
+  readonly identity: Identity;
+}
+
+export class Governor {
+  private readonly connection: BnsConnection;
+  private readonly identity: Identity;
+  private readonly address: Address;
+
+  public constructor({ connection, identity }: GovernorOptions) {
+    this.connection = connection;
+    this.identity = identity;
+    this.address = bnsCodec.identityToAddress(this.identity);
+  }
+
+  public async getElectorates(): Promise<readonly Electorate[]> {
+    const electorates = await this.connection.getElectorates();
+    return electorates.filter(({ electors }) => Object.keys(electors).some(key => key === this.address));
+  }
+
+  public async getElectionRuleId(electorateId: number): Promise<string> {
```

Comment:
This assumes there is a 1:1 mapping between an electorate and an election rule. Is this a valid assumption? If yes, it should be documented in a small comment or something.

Reply:
It doesn't assume a 1:1 mapping, it just assumes that for most purposes you're only interested in the most recent rule for a given electorate. Maybe it should instead find the rule with the version which matches the current electorate version?

Reply:
But when there are rule `(id: 1, version: 1)`, `(id: 1, version: 2)`, `(id: 2, version: 1)`, only the latest version of the first rule is returned.

> Maybe it should instead find the rule with the version which matches the current electorate version?

I don't think the versions of the rule and the electorate are necessarily in sync

Reply:
OK, clarified things with Alex. The electorates and election rules are established in the genesis block and currently there is no way to modify/add these after genesis. In principle one electorate can have more than one election rule, and when creating a proposal, the proposer is free to choose which of these to use. In practice though, we're likely to have only one election rule per electorate. I've updated the method to return an array of election rules so that the caller can select which one to use.

Reply:
> In practice though, we're likely to have only one election rule per electorate.

Okay, we can make this assumption in the caller code, where it is easy to generalize